app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb

```diff
@@ -124,17 +124,25 @@ def monitor_updates(vim, property_filter, version)
 
   def vim_connect
     host = ems.hostname
+    port = ems.port || 443
     username, password = ems.auth_user_pwd
 
+    insecure = ems.verify_ssl == OpenSSL::SSL::VERIFY_NONE
+
+    ca_file  = Tempfile.new
+    ca_file.write(ems.certificate_authority)
+    ca_file.close
```

Comment:
This is a little gross but I can't find a good way right now to pass a ca_file string rather than a filename to RbVmomi

Reply:
I think this is problematic only because you used Tempfile, which will delete the file when the `ca_file` object is GC'd.  Also tempfile creates the object with 600 permission, which may not make a difference, but if it needs to be shared across processes it _might_ be a problem when we go non-root.

I think there's a way to get a Tempfile _name_, and then you can just create it manually.  Of course it won't ever get removed, so I don't know if that's better or not.

Reply:
`Tempfile` uses `Dir::Tmpname.create` which uses its own method for creating a unique dirname so we can't call into that directly.

I didn't go the block route because the ca_file is optional and having a block which optionally yielded `nil` felt a little weird.

Reply:
They restart the http connection on exception so this Tempfile staying in scope for this method isn't good enough.  It is too bad I can't pass the Tempfile object to the RbVmomi opts so that it stayed in scope for the lifetime of the connection.  Going to have to manage the file manually and unlink it on logout

Reply:
Okay @Fryguy PTAL, I'm keeping the object in scope by assigning it to an ivar then unlinking it after logout

Reply:
This is much better 👍